pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationService.java

```diff
@@ -99,18 +99,38 @@ public String authenticate(AuthenticationDataSource authData, String authMethodN
     }
 
     public String authenticateHttpRequest(HttpServletRequest request) throws AuthenticationException {
-        // Try to validate with any configured provider
         AuthenticationException authenticationException = null;
         AuthenticationDataSource authData = new AuthenticationDataHttps(request);
-        for (AuthenticationProvider provider : providers.values()) {
+        String authMethodName = request.getHeader("Auth-Method-Name");
```

Comment:
Since this is a non-standard header, we should use a prefixed name, like `X-Pulsar-Auth-Method-Name` or `X-Pulsar-Auth-Method`

Reply:
Done in f97ccc37bbe02814434a3a1fd54e9f964fadd13c.